dulwich/protocol.py

```diff
@@ -107,6 +107,8 @@
     ]
 )
 
+DEPTH_INFINITE = 0x7FFFFFFF
```

Comment:
This is documented here: https://git-scm.com/docs/shallow

> The special depth 2147483647 (or 0x7fffffff, the largest positive number a signed 32-bit integer can contain) means infinite depth.

And can also be seen in git source `commit.h`. When `fetch --unshallow` is used, CLI git just calls `git-fetch-pack` with depth set to this value

Reply:
This seems fine, but is unrelated to the rest of the pull request?